src/main/java/org/onedatashare/server/service/UserService.java

```diff
@@ -154,17 +155,28 @@ public GlobusClient getGlobusClientFromUser(User user){
 
   public Mono<String> resetPasswordWithOld(String cookie, String oldpassword, String newpassword, String passwordConfirm){
     return getLoggedInUser(cookie).flatMap(user-> {
+      //OldPwdMatchingException oldPwd = new OldPwdMatchingException()
+      //System.out.println("oldpassword "+oldpassword);
       if(!newpassword.equals(passwordConfirm)){
         ODSLoggerService.logError("Passwords don't match.");
-        return Mono.error(new Exception("Passwords don't match."));
+        throw new  OldPwdMatchingException("Passwords don't match.");
       }else if(!user.checkPassword(oldpassword)){
         ODSLoggerService.logError("Old Password is incorrect.");
-        return Mono.error(new Exception("Old Password is incorrect."));
+       // return Mono.error(new Exception("Old Password is incorrect."));
+        throw new  OldPwdMatchingException("Old Password is incorrect.");
       }else{
+        try{
         user.setPassword(newpassword);
         userRepository.save(user).subscribe();
         ODSLoggerService.logInfo("Password reset for user " + user.getEmail() + " successful.");
         return Mono.just(user.getHash());
+        }
+        catch (RuntimeException e)
+        {
+          throw  new OldPwdMatchingException(e.getMessage());
+        }
+
+

```

Comment:
The indentation doesn't look proper. Please fix it.

Reply:
fixed indentation